doc/source/tutorial/interpolate.rst

```diff
@@ -10,38 +10,494 @@ Interpolation (:mod:`scipy.interpolate`)
 
 .. contents::
 
-There are several general interpolation facilities available in SciPy,
-for data in 1, 2, and higher dimensions:
+There are several general facilities available in SciPy for interpolation and
+smoothing for data in 1, 2, and higher dimensions. The choice of a specific
+interpolation routine depends on whether your data is
 
-- A class representing an interpolant (:class:`interp1d`) in 1-D,
-  offering several interpolation methods.
+- :ref:`One-dimensional <tutorial-interpolate_1Dsection>` (1-D);
 
-- Convenience function :func:`griddata` offering a simple interface to
-  interpolation in N dimensions (N = 1, 2, 3, 4, ...).
-  Object-oriented interface for the underlying routines is also
-  available.
+- Is given on a :ref:`structured grid <tutorial-interpolate_regular_grid_interpolator>`
+  in arbitrary (N) dimensions;
 
-- :class:`RegularGridInterpolator` provides several interpolation methods
-  on a regular grid in arbitrary (N) dimensions,
+- Is :ref:`unstructured <tutorial-interpolate_NDunstructured>`;
 
-- Functions for 1- and 2-D (smoothed) cubic-spline
-  interpolation, based on the FORTRAN library FITPACK. They are both
-  procedural and object-oriented interfaces for the FITPACK library.
+For data smoothing, :ref:`functions are provided <tutorial-interpolate_fitpack>`
+for 1- and 2-D (smoothed)
+cubic-spline interpolation, based on the FORTRAN library FITPACK. There are
+two interfaces for the FITPACK library, a procedural interface and an
+object-oriented interface.
 
-- Interpolation using radial basis functions.
+Additionally, routines are provided for interpolation / smoothing using
+:ref:`radial basis functions <tutorial-interpolate_RBF>` with several kernels.
 
 
-1-D interpolation (:class:`interp1d`)
-=====================================
+Missing data
+============
+
+Before describing various interpolators in detail, we note the (lack of) support
+for interpolation with missing data. Two popular ways of representing missing
+data are using masked arrays of the `numpy.ma` library, and encoding missing
+values as not-a-number, ``NaN``. 
+
+None of these two approaches is directly suppored in `scipy.interpolate`.
+Individual routines may offer partial support, and/or workarounds, but in
+general the library firmly adheres to the IEEE 754 semantics where a ``NaN``
+means *not-a-number*, i.e. a result of an illegal mathematical operation
+(think a division by zero), not *missing*.
+
+
+.. _tutorial-interpolate_1Dsection:
+
+1-D interpolation
+=================
+
+Piecewise linear interpolation
+------------------------------
+
+If all you need is a linear (a.k.a. broken line) interpolation, you can use
```

Comment:
There are, too many of them in fact :-). `interp1d(x, y, kind="linear")` or `make_interp_spline(x, y, k=1)` [which is what `interp1d(x, y, kind="linear")`, a third equivalent way, delegates to under the hood.]  Do you think any of these are helpful this early in the page?

Reply:
Hmm I thought it would be simpler than cubic, but it's not. Up to you - is it worth mentioning the `k` parameter of `make_interp_spline`? You talk about the degree of B-splines `k`, but it's not passed as an argument anywhere.

Reply:
@vnmabus no, it is not similar. `np.interp` is generally fastest, and is in fact recommended. `interp1d` has the most overhead and is not recommended. `make_interp_spline(...., k=1)`  should be comparable to `np.interp` if somewhat slower. This is probably useful is limited contexts: when you want flexibility in `k=...` or if you insist on passing around a linearly interpolating callable object.